src/wp-includes/class-wp-user.php

```diff
@@ -740,7 +767,8 @@ public function remove_cap( $cap ) {
 	 */
 	public function remove_all_caps() {
 		global $wpdb;
-		$this->caps = array();
+		$this->caps        = array();
+		$this->loaded_caps = false;
 		delete_user_meta( $this->ID, $this->cap_key );
 		delete_user_meta( $this->ID, $wpdb->get_blog_prefix() . 'user_level' );
 		$this->get_role_caps();
```

Comment:
This implementation looks a bit inconsistent with that from the other methods, and it is doing manually what `load_capability_data()` does. So I think it makes sense to use it:

```suggestion
		delete_user_meta( $this->ID, $this->cap_key );
		delete_user_meta( $this->ID, $wpdb->get_blog_prefix() . 'user_level' );
		$this->loaded_caps = false;
		$this->load_capability_data();
```

By calling `load_capability_data()` here, `$this->caps` will be automatically set to the empty array (if deleting the user metadata succeeded), and then `$this->get_role_caps()` will be called.